java_src/src/main/java/com/google/crypto/tink/integration/hcvault/HcVaultKmsClient.java

```diff
@@ -0,0 +1,114 @@
+package com.google.crypto.tink.integration.hcvault;
+
+import com.bettercloud.vault.Vault;
+import com.bettercloud.vault.VaultConfig;
+import com.bettercloud.vault.VaultException;
+import com.google.auto.service.AutoService;
+import com.google.crypto.tink.Aead;
+import com.google.crypto.tink.KmsClient;
+import com.google.crypto.tink.subtle.Validators;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+
+/**
+ * An implementation of {@link KmsClient} for <a
+ * href="https://www.vaultproject.io/docs/secrets/transit">Vault Transit Secrets Engine</a>..
+ */
+@AutoService(KmsClient.class)
+public class HcVaultKmsClient implements KmsClient {
+
+  public static final String PREFIX = "hcvault://";
+
+  private String keyUri;
+  private Vault vault;
+
+  public HcVaultKmsClient() {}
+
+  /** Constructs a specific HcVaultKmsClient that is bound to a single key identified by {@code uri}. */
+  public HcVaultKmsClient(String uri) {
+    if (!uri.toLowerCase().startsWith(PREFIX)) {
+      throw new IllegalArgumentException("key URI must starts with " + PREFIX);
+    }
+    this.keyUri = uri;
+  }
+
+  /**
+   * @return @return true either if this client is a generic one and uri starts with {@link
+   * HcVaultKmsClient#PREFIX}, or the client is a specific one that is bound to the key identified
+   * by {@code uri}.
+   */
+  @Override
+  public boolean doesSupport(String uri) {
+    if (this.keyUri != null && this.keyUri.equals(uri)) {
+      return true;
+    }
+    return this.keyUri == null && uri.toLowerCase().startsWith(PREFIX);
+  }
+
+  /**
+   * Loads Vault config with the host provided in the {@code credentialPath}.
+   *
+   * <p>If {@code credentialPath} is null, loads address from environment variables.</p>
+   * <p>
+   * All other configuration elements will also be read from environment variables.
+   */
+  @Override
+  public KmsClient withCredentials(String credentialPath) throws GeneralSecurityException {
+    if (!credentialPath.toLowerCase().startsWith(PREFIX)) {
+      throw new IllegalArgumentException("key URI must starts with " + PREFIX);
+    }
+
+    try {
+      URI uri = new URI(credentialPath);
+      VaultConfig config = new VaultConfig()
```

Comment:
Without a token, does this actually work?

Perhaps we should follow https://github.com/google/tink/blob/master/go/integration/hcvault/hcvault_client.go#L52.

Reply:
VaultConfig will always look for a token in an environment variable named `VAULT_TOKEN`. I can add the token as you suggested, However I wasn't really sure that `withCredentials` made sense for a vault implementation to begin with.